File: solutions/1560. Most Visited Sector in a Circular Track/1560.java
class Solution {
  public List<Integer> mostVisited(int n, int[] rounds) {
    // 1. If start <= end, [start, end] is the most visited.
    //
    //      s --------- n
    // 1 -------------- n
    // 1 ------ e
    //
    // 2. If start > end, [1, end] and [start, n] are the most visited.
    //
    //             s -- n
    // 1 -------------- n
    // 1 ------ e
    final int start = rounds[0];
    final int end = rounds[rounds.length - 1];
    List<Integer> ans = new ArrayList<>();
    for (int i = 1; i <= n; ++i)
      if (start <= end) {
        if (start <= i && i <= end)
          ans.add(i);
      } else { // start > end
        if (i >= start || i <= end)
          ans.add(i);
      }
    return ans;
  }
}

File: solutions/1560. Most Visited Sector in a Circular Track/1560.py
class Solution:
  def mostVisited(self, n: int, rounds: List[int]) -> List[int]:
    # Case 1: if start <= end, [start, end] is the most visited.
    #
    #      s --------- n
    # 1 -------------- n
    # 1 ------ e
    #
    # Case 2: if start > end, [1, end] and [start, n] are the most visited.
    #
    #             s -- n
    # 1 -------------- n
    # 1 ------ e
    start = rounds[0]
    end = rounds[-1]
    if start <= end:
      return range(start, end + 1)
    return list(range(1, end + 1)) + list(range(start, n + 1))
